chain-config/src/validator_rules.rs

```diff
@@ -21,10 +24,91 @@ pub trait ValidatorRulesModule {
         }
     }
 
+    #[endpoint(register)]
+    fn register(&self, new_validator: ValidatorInfo<Self::Api>) {
+        self.require_setup_complete();
+        self.require_validator_not_registered(&new_validator.bls_key);
+
+        let max_number_of_validators = self.sovereign_config().get().max_validators;
+        let last_bls_key_id = self.last_bls_key_id().get();
+        let current_bls_key_id = &last_bls_key_id + 1u32;
+
+        require!(
+            current_bls_key_id <= max_number_of_validators,
+            VALIDATOR_RANGE_EXCEEDED
+        );
+
+        self.last_bls_key_id().set(current_bls_key_id.clone());
+        self.id_to_bls_key_mapper(&current_bls_key_id)
+            .set(new_validator.bls_key.clone());
+        self.bls_key_to_id_mapper(&new_validator.bls_key)
+            .set(current_bls_key_id);
+
+        self.register_event(
+            &new_validator.address,
+            &new_validator.bls_key,
+            &new_validator.egld_stake,
+            &new_validator.token_stake,
+        );
+    }
+
+    #[endpoint(unregister)]
+    fn unregister(&self, validator_info: ValidatorInfo<Self::Api>) {
+        self.require_setup_complete();
+        self.require_validator_registered(&validator_info.bls_key);
+
+        let min_number_of_validators = self.sovereign_config().get().min_validators;
+        let last_bls_key_id = self.last_bls_key_id().get();
+        let current_bls_key_id = &last_bls_key_id - 1u32;
+
+        require!(
+            current_bls_key_id >= min_number_of_validators,
```

Comment:
This is not really ok. Validators should be able to unregister.

Reply:
There is no need for an event. The min number of validators will be checked in other places.